conformance/tests/specialtypes_promotions.py

```diff
@@ -2,6 +2,8 @@
 Tests "type promotions" for float and complex when they appear in annotations.
 """
 
+from typing import assert_type
+
 # Specification: https://typing.readthedocs.io/en/latest/spec/special-types.html#special-cases-for-float-and-complex
 
 v1: float = 1
```

Comment:
This line implies that a `float` variable annotation also means `float | int`, but it's not super clear (it could also be interpreted as just a coercion of literals based on the type context). I would slightly prefer to see an explicit test of the type system rules, in the form `x: int = 1; y: float = x`.

Reply:
Good point, added some more variants.